### DISCUSSION TITLE
issue: getFieldState from useFormContext does not have latest error state without referencing formState also

### DISCUSSION THREAD

**Discussion by jbgrunewald** [react-hook-form/react-hook-form#11863] [Q&A]
### Version Number

7.49.3

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/crimson-dew-znxv25?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvh3uy7t00063b6jpx47mffe%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvh3uy7t00023b6j8dxz4prp%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvh3uy7t00033b6jg745aow0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvh3uy7t00053b6jaymqmboh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvh3uy7t00023b6j8dxz4prp%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvh3uy7t00013b6jea4oxwei%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvh3wsdu00023b6icj2ah9r8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clvh3xmfm002m3b6ipasuwv0i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A19%252C%2522startColumn%2522%253A15%252C%2522endLineNumber%2522%253A19%252C%2522endColumn%2522%253A15%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvh3uy7t00023b6j8dxz4prp%2522%252C%2522activeTabId%2522%253A%2522clvh3xmfm002m3b6ipasuwv0i%2522%257D%252C%2522clvh3uy7t00053b6jaymqmboh%2522%253A%257B%2522id%2522%253A%2522clvh3uy7t00053b6jaymqmboh%2522%252C%2522activeTabId%2522%253A%2522clvh4aees00kv3b6i6a9y2qty%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clvh4aees00kv3b6i6a9y2qty%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clvh3uy7t00033b6jg745aow0%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvh3uy7t00033b6jg745aow0%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to reproduce

1.Enter 3 or more characters into the text input provided
2. Observe that no error message or error styling is applied (resolver has 2 character limit)
3. Add the formState to the fields from useFormContext like the example change below

```
 const { control, getFieldState, formState } = useFormContext<T>();
  const { error } = getFieldState(id);
  const hasErrors = !!error;
  const formError = formState.errors;
```

### Expected behaviour

I expect the getFieldState to accurately report the error state regardless of referencing the formState. Perhaps I'm misunderstanding something, but it seems that the getFieldState is not causing the rerender correctly in a context based setup.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> it seems that the getFieldState is not causing the rerender correctly in a context based setup.

`getFieldState` doesn't trigger re-render, it only retrieve the latest form state for the input
